### DISCUSSION TITLE
Strongswan IPSec TP-Link TL-MR3020

### DISCUSSION THREAD

**Discussion by PatrickKalka-SeriousByte** [strongswan/strongswan#373] [Q&A] [answered]
Hello Guys,

I'm not realy familiar with VPN, so I'm totaly sorry for asking obvious Questions.

All Mobile Devices should be able to interact with Synology by using the VPN Connection of MR3020. And of cause use the Internet starting at Speedport.
https://www.seriousbyte.de/fileadmin/user_upload/VPN-Setup/VPN-Setup.png

These are the open ports at Speedport.
https://www.seriousbyte.de/fileadmin/user_upload/VPN-Setup/Port-Forwarding.PNG

MR3020 VPN configuration
https://www.seriousbyte.de/fileadmin/user_upload/VPN-Setup/MR3020-VPNSettings.PNG
And this is all Documentation about MR3020 https://www.tp-link.com/de/support/faq/1891/

`At reboot:`
> sudo iptables -t nat -A POSTROUTING -s 10.2.0.0/24 -o wlan0 -j MASQUERADE
sudo iptables -t nat -A POSTROUTING -s 10.2.0.0/24 -o eth0 -j MASQUERADE
sudo iptables -A FORWARD -s 10.2.0.0/24 -j ACCEPT
sudo iptables -A INPUT -p udp --dport 500 -j ACCEPT
sudo iptables -A INPUT -p udp --dport 4500 -j ACCEPT

`sudo nano /etc/ipsec.conf`
> config setup
    cachecrls=yes
    uniqueids=yes
conn Mr3020
    keyexchange=ikev1
    authby=psk
    left=%defaultroute
    leftsubnet=192.168.2.0/24
    right=%any
    rightsubnet=10.2.0.0/24
    auto=add
    esp=aes128-sha1-modp1024

`sudo nano /etc/sysctl.conf`
> net.ipv4.ip_forward=1
net.ipv4.conf.all.accept_redirects = 0
net.ipv4.conf.all.send_redirects = 0

`sudo nano /etc/ipsec.secrets`
> include /var/lib/strongswan/ipsec.secrets.inc
: PSK "myKey"

`sudo journalctl | grep charon`
> Mai 05 18:37:12 VPNServer charon[16772]: 06[NET] received packet: from 84.136.198.220[500] to 192.168.2.8[500] (184 bytes)
Mai 05 18:37:12 VPNServer charon[16772]: 06[ENC] parsed ID_PROT request 0 [ SA V V V V V ]
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] received NAT-T (RFC 3947) vendor ID
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] received NAT-T (RFC 3947) vendor ID
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] received draft-ietf-ipsec-nat-t-ike-02 vendor ID
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] received draft-ietf-ipsec-nat-t-ike-02 vendor ID
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] received draft-ietf-ipsec-nat-t-ike-02\n vendor ID
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] received draft-ietf-ipsec-nat-t-ike-02\n vendor ID
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] received draft-ietf-ipsec-nat-t-ike-00 vendor ID
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] received draft-ietf-ipsec-nat-t-ike-00 vendor ID
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] received DPD vendor ID
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] received DPD vendor ID
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] 84.136.198.220 is initiating a Main Mode IKE_SA
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] 84.136.198.220 is initiating a Main Mode IKE_SA
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] IKE_SA (unnamed)[5] state change: CREATED => CONNECTING
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] IKE_SA (unnamed)[5] state change: CREATED => CONNECTING
Mai 05 18:37:12 VPNServer charon[16772]: 06[CFG] selected proposal: IKE:AES_CBC_128/HMAC_SHA1_96/PRF_HMAC_SHA1/MODP_2048
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] sending XAuth vendor ID
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] sending XAuth vendor ID
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] sending DPD vendor ID
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] sending DPD vendor ID
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] sending NAT-T (RFC 3947) vendor ID
Mai 05 18:37:12 VPNServer charon[16772]: 06[IKE] sending NAT-T (RFC 3947) vendor ID
Mai 05 18:37:12 VPNServer charon[16772]: 06[ENC] generating ID_PROT response 0 [ SA V V V ]
Mai 05 18:37:12 VPNServer charon[16772]: 06[NET] sending packet: from 192.168.2.8[500] to 84.136.198.220[500] (136 bytes)
Mai 05 18:37:12 VPNServer charon[16772]: 07[NET] received packet: from 84.136.198.220[500] to 192.168.2.8[500] (356 bytes)
Mai 05 18:37:12 VPNServer charon[16772]: 07[ENC] parsed ID_PROT request 0 [ KE No NAT-D NAT-D ]
Mai 05 18:37:12 VPNServer charon[16772]: 07[IKE] local host is behind NAT, sending keep alives
Mai 05 18:37:12 VPNServer charon[16772]: 07[IKE] local host is behind NAT, sending keep alives
Mai 05 18:37:12 VPNServer charon[16772]: 07[IKE] remote host is behind NAT
Mai 05 18:37:12 VPNServer charon[16772]: 07[IKE] remote host is behind NAT
Mai 05 18:37:12 VPNServer charon[16772]: 07[ENC] generating ID_PROT response 0 [ KE No NAT-D NAT-D ]
Mai 05 18:37:12 VPNServer charon[16772]: 07[NET] sending packet: from 192.168.2.8[500] to 84.136.198.220[500] (372 bytes)
Mai 05 18:37:13 VPNServer charon[16772]: 14[NET] received packet: from 84.136.198.220[4500] to 192.168.2.8[4500] (76 bytes)
Mai 05 18:37:13 VPNServer charon[16772]: 14[ENC] parsed ID_PROT request 0 [ ID HASH ]
Mai 05 18:37:13 VPNServer charon[16772]: 14[CFG] looking for pre-shared key peer configs matching 192.168.2.8...84.136.198.220[192.168.178.54]
Mai 05 18:37:13 VPNServer charon[16772]: 14[CFG] selected peer config "Mr3020"
Mai 05 18:37:13 VPNServer charon[16772]: 14[IKE] IKE_SA Mr3020[5] established between 192.168.2.8[192.168.2.8]...84.136.198.220[192.168.178.54]
Mai 05 18:37:13 VPNServer charon[16772]: 14[IKE] IKE_SA Mr3020[5] established between 192.168.2.8[192.168.2.8]...84.136.198.220[192.168.178.54]
Mai 05 18:37:13 VPNServer charon[16772]: 14[IKE] IKE_SA Mr3020[5] state change: CONNECTING => ESTABLISHED
Mai 05 18:37:13 VPNServer charon[16772]: 14[IKE] IKE_SA Mr3020[5] state change: CONNECTING => ESTABLISHED
Mai 05 18:37:13 VPNServer charon[16772]: 14[IKE] scheduling reauthentication in 9784s
Mai 05 18:37:13 VPNServer charon[16772]: 14[IKE] scheduling reauthentication in 9784s
Mai 05 18:37:13 VPNServer charon[16772]: 14[IKE] maximum IKE_SA lifetime 10324s
Mai 05 18:37:13 VPNServer charon[16772]: 14[IKE] maximum IKE_SA lifetime 10324s
Mai 05 18:37:13 VPNServer charon[16772]: 14[ENC] generating ID_PROT response 0 [ ID HASH ]
Mai 05 18:37:13 VPNServer charon[16772]: 14[NET] sending packet: from 192.168.2.8[4500] to 84.136.198.220[4500] (76 bytes)
Mai 05 18:37:13 VPNServer charon[16772]: 15[NET] received packet: from 84.136.198.220[4500] to 192.168.2.8[4500] (92 bytes)
Mai 05 18:37:13 VPNServer charon[16772]: 15[ENC] parsed INFORMATIONAL_V1 request 3566495395 [ HASH N(INITIAL_CONTACT) ]
Mai 05 18:37:15 VPNServer charon[16772]: 10[NET] received packet: from 84.136.198.220[4500] to 192.168.2.8[4500] (428 bytes)
Mai 05 18:37:15 VPNServer charon[16772]: 10[ENC] parsed QUICK_MODE request 3853167803 [ HASH SA No KE ID ID ]
Mai 05 18:37:15 VPNServer charon[16772]: 10[IKE] no matching CHILD_SA config found for 192.168.1.0/24 === 10.2.0.0/24
Mai 05 18:37:15 VPNServer charon[16772]: 10[IKE] no matching CHILD_SA config found for 192.168.1.0/24 === 10.2.0.0/24
Mai 05 18:37:15 VPNServer charon[16772]: 10[IKE] queueing INFORMATIONAL task
Mai 05 18:37:15 VPNServer charon[16772]: 10[IKE] queueing INFORMATIONAL task
Mai 05 18:37:15 VPNServer charon[16772]: 10[IKE] activating new tasks
Mai 05 18:37:15 VPNServer charon[16772]: 10[IKE] activating new tasks
Mai 05 18:37:15 VPNServer charon[16772]: 10[IKE]   activating INFORMATIONAL task
Mai 05 18:37:15 VPNServer charon[16772]: 10[IKE]   activating INFORMATIONAL task
Mai 05 18:37:15 VPNServer charon[16772]: 10[ENC] generating INFORMATIONAL_V1 request 810055136 [ HASH N(INVAL_ID) ]
Mai 05 18:37:15 VPNServer charon[16772]: 10[NET] sending packet: from 192.168.2.8[4500] to 84.136.198.220[4500] (76 bytes)
Mai 05 18:37:15 VPNServer charon[16772]: 10[IKE] activating new tasks
Mai 05 18:37:15 VPNServer charon[16772]: 10[IKE] activating new tasks
Mai 05 18:37:15 VPNServer charon[16772]: 10[IKE] nothing to initiate
Mai 05 18:37:15 VPNServer charon[16772]: 10[IKE] nothing to initiate
Mai 05 18:37:25 VPNServer charon[16772]: 14[NET] received packet: from 84.136.198.220[4500] to 192.168.2.8[4500] (428 bytes)
Mai 05 18:37:25 VPNServer charon[16772]: 14[IKE] received retransmit of request with ID 3853167803, but no response to retransmit
Mai 05 18:37:25 VPNServer charon[16772]: 14[IKE] received retransmit of request with ID 3853167803, but no response to retransmit
Mai 05 18:37:35 VPNServer charon[16772]: 08[NET] received packet: from 84.136.198.220[4500] to 192.168.2.8[4500] (428 bytes)
Mai 05 18:37:35 VPNServer charon[16772]: 08[IKE] received retransmit of request with ID 3853167803, but no response to retransmit
Mai 05 18:37:35 VPNServer charon[16772]: 08[IKE] received retransmit of request with ID 3853167803, but no response to retransmit
Mai 05 18:37:35 VPNServer charon[16772]: 07[IKE] sending keep alive to 84.136.198.220[4500]
Mai 05 18:37:35 VPNServer charon[16772]: 07[IKE] sending keep alive to 84.136.198.220[4500]
Mai 05 18:37:55 VPNServer charon[16772]: 05[IKE] sending keep alive to 84.136.198.220[4500]
Mai 05 18:37:55 VPNServer charon[16772]: 05[IKE] sending keep alive to 84.136.198.220[4500]
Mai 05 18:38:13 VPNServer charon[16772]: 10[NET] received packet: from 84.136.198.220[4500] to 192.168.2.8[4500] (92 bytes)
Mai 05 18:38:13 VPNServer charon[16772]: 10[ENC] parsed INFORMATIONAL_V1 request 4118340783 [ HASH N(DPD) ]
Mai 05 18:38:13 VPNServer charon[16772]: 10[IKE] queueing ISAKMP_DPD task
Mai 05 18:38:13 VPNServer charon[16772]: 10[IKE] queueing ISAKMP_DPD task
Mai 05 18:38:13 VPNServer charon[16772]: 10[IKE] activating new tasks
Mai 05 18:38:13 VPNServer charon[16772]: 10[IKE] activating new tasks
Mai 05 18:38:13 VPNServer charon[16772]: 10[IKE]   activating ISAKMP_DPD task
Mai 05 18:38:13 VPNServer charon[16772]: 10[IKE]   activating ISAKMP_DPD task
Mai 05 18:38:13 VPNServer charon[16772]: 10[ENC] generating INFORMATIONAL_V1 request 1596706606 [ HASH N(DPD_ACK) ]
Mai 05 18:38:13 VPNServer charon[16772]: 10[NET] sending packet: from 192.168.2.8[4500] to 84.136.198.220[4500] (92 bytes)
Mai 05 18:38:13 VPNServer charon[16772]: 10[IKE] activating new tasks
Mai 05 18:38:13 VPNServer charon[16772]: 10[IKE] activating new tasks
Mai 05 18:38:13 VPNServer charon[16772]: 10[IKE] nothing to initiate
Mai 05 18:38:13 VPNServer charon[16772]: 10[IKE] nothing to initiate
Mai 05 18:38:34 VPNServer charon[16772]: 11[IKE] sending keep alive to 84.136.198.220[4500]
Mai 05 18:38:34 VPNServer charon[16772]: 11[IKE] sending keep alive to 84.136.198.220[4500]

I think, after playing with the esp-Konfiguration, I got an connection, but it MR3020 says still "VPN down" and i'm not sure what the log is telling me. 

I hope you could guide the way.

**Comment by Thermi** [comment]
> > leftsubnet=192.168.2.0/24
> [...]
> > Mai 05 18:37:15 VPNServer charon[16772]: 10[IKE] no matching CHILD_SA config found for 192.168.1.0/24 === 10.2.0.0/24
> > Mai 05 18:37:15 VPNServer charon[16772]: 10[IKE] no matching CHILD_SA config found for 192.168.1.0/24 === 10.2.0.0/24
> [...]


192.168.2.0/24 != 192.168.1.0/24

Evil typo strikes again.

**Reply by PatrickKalka-SeriousByte** [reply]
thanks for your quick answer. 
According to my selected IPs, (https://www.seriousbyte.de/fileadmin/user_upload/VPN-Setup/VPN-Setup.png) shouldn't left be 192.168.2.0 ?

**Reply by Thermi** [reply]
If you want it to work without collisions, you should just dice roll the subnets you use. Yes, left should be 192.168.2.0/24, if you keep those. If you SNAT/MASQUERADE connections on the MR320 side, you need to use the subnet/IP you're SNATing the connections to. If you can, you should disable that SNAT though.

**Comment by PatrickKalka-SeriousByte** [comment, marked as answer]
https://www.seriousbyte.de/fileadmin/user_upload/VPN-Setup/MR3020-VPNSettings.PNG

As usual, the Problem was 30 cm infront of the screen. The remote IP Address has to be 192.168.2.0 instead of 10.2.0.1  And every thing works like charm.
